AEPMessaging/Sources/Messaging.swift

```diff
@@ -617,8 +617,16 @@ public class Messaging: NSObject, Extension {
             case .inapp:
                 Log.trace(label: MessagingConstants.LOG_TAG, "Updating in-app message definitions for surfaces \(newRules.compactMap { $0.key.uri }).")
 
+                /// MOB-21852 - iam items are returned in reverse priority order, so we need to flip the
+                /// order of the array prior to saving them and hydrating the rules engine.  this allows the
+                /// highest priority item to be shown first when rules engine evaluates top-down
+                var newRulesReordered: [Surface: [LaunchRule]] = [:]
+                for (surface, surfaceRules) in newRules {
+                    newRulesReordered[surface] = surfaceRules.reversed()
```

Comment:
would it make sense to reverse the priority before the IAM rules are cached? this would remove the need to reverse the order when reading from the cache.

Reply:
discussed this offline. the android implementation will be following this as well.